src/include/kdb.h

```diff
@@ -747,6 +728,12 @@ void krb5_db_free_authdata_info(krb5_context context, void *ad_info);
 void
 krb5_dbe_sort_key_data(krb5_key_data *key_data, size_t key_data_length);
 
+krb5_error_code
+krb5_db_issue_pac(krb5_context context, unsigned int flags,
+                  krb5_const_principal client_princ, krb5_db_entry *client,
+                  krb5_timestamp authtime, krb5_boolean is_cross_realm,
+                  krb5_pac *pac_in, krb5_pac *pac_out);
```

Comment:
Do we want pac_in/pac_out or should the method just modify the PAC in place with krb5_pac_add_buffer()?

Reply:
Currently the IPA code always makes a new PAC.  The worry seems to be that resigning with a different algorithm will take up a different amount of space, which would cause problems.

Reply:
But IPA wouldn't be responsible for signing the PAC with the new interface, would it?

Reply:
This is true.  We could move that logic into krb5.

Reply:
A larger, currently undocumented, problem is that some fields need modified and we don't have a way to do that.   For instance, the transited services KRB5_PAC_DELEGATION_INFO: it needs to be pulled from the old PAC, updated, and shoved into the new one.  So it seems to me that we either need to either make reissuing the responsibility of the plugin, or we need field update and deletion methods for krb5_pacs.  I don't really have a preference between the two, but the second is more work so I'll defer it until later.

Reply:
Ideally delegation info would be the responsibility of the KDC, not the KDB module.

Reply:
More generally, from context I think you're envisioning that pac_in is the PAC from the header TGT or evidence ticket.  The design I understood from Isaac is that the KDC creates a minimal PAC for the KDB module to extend.  If the KDB module does nothing, the KDC still issues the minimal PAC.  The PAC from an incoming ticket can be a separate parameter.